File: 0001-two-sum/0001-two-sum.go
func twoSum(nums []int, target int) []int {
    comps := make(map[int]int, len(nums))
    for i, n := range nums {
        if v, ok := comps[target-n]; ok {
            return []int{v, i}
        }
        comps[n] = i
    }
    return nil
}

File: 0001-two-sum/0001-two-sum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
let twoSum = function(nums, target) {
    const valToIdx = new Map();
    for (let i = 0; i < nums.length; i++) {
        let complement = target - nums[i];
        if (valToIdx.has(complement)) {
            return [valToIdx.get(complement), i];
        }
        valToIdx.set(nums[i], i);
    }
    return [];
};